### PR TITLE
Change balancer ring creation to block-based memory allocation

### PR DESCRIPTION
**Pull request by levon-a-akopian**
… memory for the ring is allocated in blocks

### CODE REVIEW

**Comment by ol-imorozko** [comment]
Please use correct commit message conventions, that is, the first line should not be that long.
Smth like:
```
Change balancer ring creation to block-based memory allocation

This improves memory management efficiency by allocating ring memory in blocks.
```

File: README.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ docker pull yanetplatform/builder
 Add alias for run commands on docker:
 ```
 alias yanet-builder="docker run --rm -it -v /run/yanet:/run/yanet -v \$PWD:/project yanetplatform/builder"
+(for Mac: alias yanet-builder="docker run --platform linux/amd64 --rm -it -v /run/yanet:/run/yanet -v \$PWD:/project yanetplatform/builder")
```
Comment (ol-imorozko) [inline README.md:36]: This is not relevant change, please create another pull request

File: controlplane/balancer.cpp [hunk 2]
```diff
@@ -668,14 +695,13 @@ void balancer_t::update_service(const balancer::generation_config_t& generation_
 
 			for (const auto& [real_ip, real_port, weight] : reals)
 			{
-				std::tuple<std::string, balancer::service_key_t, balancer::real_key_t> key = {module_name, {virtual_ip, proto, virtual_port}, {real_ip, real_port}};
+		    	balancer::real_key_global_t key = {module_name, {virtual_ip, proto, virtual_port}, {real_ip, real_port}};
```
Comment (ol-imorozko) [inline controlplane/balancer.cpp:698]: `clang-format`

File: controlplane/balancer.h [hunk 3]
```diff
@@ -26,9 +26,11 @@ using service_key_t = std::tuple<common::ip_address_t,
 using service_counter_key_t = std::tuple<std::string, ///< module_name
                                          service_key_t>;
 
-using real_counter_key_t = std::tuple<std::string, ///< module_name
-                                      service_key_t,
-                                      real_key_t>;
+using real_key_global_t = std::tuple<std::string, ///< module_name
+                                    service_key_t,
+                                    real_key_t>;
```
Comment (TheRandomCharacter) [inline controlplane/balancer.h:31]: Is `tuple` really needed here? We should prefer `struct` to `tuple` in cases where we don't need to iterate over members.

If resorting to using tuple is unavoidable generous usage of type aliases makes code more readable:
```
using ModuleName = std::string
std::tuple<ModuleName>
```
is better than
```
std::tuple<std::string ///> module_name
>
```

File: README.md (hunk 1 above)
Comment (GeorgyKirichenko) [reply README.md:36]: Yes, this is our overall politic to split non-relevant changes even if changes are small.

File: controlplane/balancer.cpp [hunk 4]
```diff
@@ -354,6 +354,23 @@ void balancer_t::reload_after()
...
+			std::lock_guard<std::mutex> guard(reals_enabled_mutex);
+			// At the point real_updates and real_reload_updates
+			// may contain different sets of reals updates depending
+			// on update, flush and reload operation interposition.
+			// Besides the possibility that real_reload_updates is
+			// less than its counterpart it should be enough to
+			// flush only the first one and then reset both of them
+			// as real_reload_updates initial state already contains
+			// all changes was made just before the reload operation
+			// started. So we just want to rewu
```
Comment (GeorgyKirichenko) [inline controlplane/balancer.cpp:368]: I feel the message is unfinished

**Review by GeorgyKirichenko — CHANGES_REQUESTED** [review]
I feel the PR is mostly about #35 so please add corresponding line into the commit message like "Closed #35 or Relates #35"
Also please find some consideration about bellow:
 - Please move all irrelevant changes about incremental real update into separate PR as there is an #35
 -  Also #35 suggests that we could eliminate the second copy of real mapping ring as each service will have its reserved space inside the mapping but I could not see here any corresponding changes.

File: README.md (hunk 1 above)
Comment (levon-a-akopian) [reply README.md:36]: Ok

**Review by GeorgyKirichenko — APPROVED** [review]

**Pull request merged**